File: cpp/206-Reverse-Linked-List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* rev = new ListNode(0);
        ListNode* ptr = head;
        
        while(ptr) {
            auto temp = ptr;
            ptr=ptr->next;
            temp-> next = rev->next;
            rev->next = temp;
        }
        
        return rev->next;
    }
};

File: csharp/206-Reverse-Linked-List.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode ReverseList(ListNode head) {
        ListNode prev = null, cur = head;
        
        while(cur != null) {
            var temp = cur.next;
           
            cur.next = prev;
            prev = cur;
            cur = temp;
        }
        
        return prev;
    }
}